spec/unit/pops/types/type_calculator_spec.rb

```diff
@@ -208,6 +208,12 @@ def types
       end
     end
 
+    context 'sensitive' do
+      it 'translates to PSensitiveType' do
+        expect(calculator.infer(PSensitiveType::Sensitive.new("hunter2")).class).to eq(PSensitiveType)
+      end
+    end
```

Comment:
This is indeed a tricky problem. Generalizing the type removes all details, but it also makes it impossible to use more precise typing when say, Sensitive[String[6]] or Sensitive[Enum[red,blue, green]]. Would it be enough to present the generalized type T when the type is stringified? It could perhaps output the type as Stringified[Redacted] or somesuch.

Another alternative would be to support a Sensitive subtype that the inference uses. This type does not reveal the detailed type, but the type system would have access to the detailed information. Done that way, user entered types like `Sensitive[Enum[blue,red,green]]` would print the details, but an inferred type would not.

Indeed a hard problem.